gapic-generator/templates/default/service/client/_client.erb

```diff
@@ -21,22 +21,25 @@ class <%= service.client_name %>
   # @private
   attr_reader :<%= service.stub_name %>
 
-  # The default address of the service.
-  SERVICE_ADDRESS = "<%= service.client_address %>"
-
-  # The default port of the service.
-  DEFAULT_SERVICE_PORT = <%= service.client_port %>
-
-  # rubocop:disable Style/MutableConstant
-
-  # The default set of gRPC interceptors.
-  GRPC_INTERCEPTORS = []
-
-  # rubocop:enable Style/MutableConstant
-
-  DEFAULT_TIMEOUT = 30
-
+  ##
+  # Configuration for the <%= service.name %> <%= service.client_name %> API.
+  #
+  def self.configure
+    @configure ||= <%= indent_tail render(partial: "service/client/config", locals: { service: service }), 4 %>
+    yield @configure if block_given?
+    @configure
+  end
 
+  ##
+  # Configure the <%= service.name %> <%= service.client_name %> instance.
+  #
+  # The configuration is set to the derived mode, meaning that values can be changed,
+  # but structural changes (adding new fields, etc.) are not allowed. Structural changes
+  # should be made on {<%= service.client_name %>.configure}.
+  def configure
```

Comment:
By allowing an instance object's configuration to be updated, we have unintentionally opened the Client object up to more name conflicts. If the API has a `configure` method  then this method will be overwritten.

What if we allowed the instance object to be passed to the global configure?

```ruby
# Set global defaults
Google::Cloud::Vision::V1::ImageAnnotator::Client.configure do |config|
  config.timeout = 120
end

# Override the defaults when creating an instance
client = Google::Cloud::Vision::V1::ImageAnnotator::Client.new do |config|
  config.timeout = 90
end

# Configure the instance defaults by passing the client to Client.configure
Google::Cloud::Vision::V1::ImageAnnotator::Client.configure client do |config|
  config.timeout = 60
end
```

Reply:
For example, if an API has a configure method, then calling `Client#configure` will make the RPC API call. But passing the client instance object to `Client.configure` would still allow the client to be configured without calling `Client#configure`.